src/core/ignore/ClineIgnoreController.ts

```diff
@@ -58,17 +58,41 @@ export class ClineIgnoreController {
 	}
 
 	/**
-	 * Load custom patterns from .clineignore if it exists
+	 * Load custom patterns from .clineignore if it exists.
+	 * Supports "!include <filename>" to load additional ignore patterns from other files.
 	 */
 	private async loadClineIgnore(): Promise<void> {
 		try {
 			// Reset ignore instance to prevent duplicate patterns
 			this.ignoreInstance = ignore()
 			const ignorePath = path.join(this.cwd, ".clineignore")
+
 			if (await fileExistsAtPath(ignorePath)) {
-				const content = await fs.readFile(ignorePath, "utf8")
-				this.clineIgnoreContent = content
-				this.ignoreInstance.add(content)
+				const rawContent = await fs.readFile(ignorePath, "utf8")
+				this.clineIgnoreContent = rawContent
+
+				let combinedContent = ""
+
```

Comment:
it is a good idea to first check for the existence of "!include" before proceeding

Reply:
Thank you for confirming this. I agree with you. Corrected.
https://github.com/cline/cline/blob/0ebc48a7ce17317d7be45a5ff8e24c395ac876c6/src/core/ignore/ClineIgnoreController.ts#L86-L96